src/features/rewards/toggleTips/index.tsx

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import { getLocale } from '../../../helpers'
+
+import {
+  StyledEnableTipsSection,
+  StyledEnableTipsInner,
+  StyledText,
+  StyledProviderImg,
+  StyledEnableTips,
+  StyledProviderName,
+  StyledToggleOuter,
+  StyledToggleInner,
+  StyledThumbsUpIcon
+} from './style'
+import { Toggle } from '../../../components'
+
+export type Provider = 'twitter' | 'youtube' | 'twitch'
+
+export interface Props {
+  id?: string
+  provider?: Provider
+  onToggleTips: () => void
+  tipsEnabled?: boolean
+}
+
+const thumbsUpIcon = require('./assets/thumbsUp')
+const youtubeIcon = require('./assets/youtube')
+const twitterIcon = require('./assets/twitter')
+const twitchIcon = require('./assets/twitch')
+
+export default class ToggleTips extends React.PureComponent<Props, {}> {
+  getProviderImg (provider: Provider) {
+    switch (provider) {
+      case 'youtube':
+        return (<span>{youtubeIcon}</span>)
+      case 'twitter':
+        return (<span>{twitterIcon}</span>)
+      case 'twitch':
+        return (<span>{twitchIcon}</span>)
+    }
+  }
+
+  getProviderName (provider: Provider) {
+    return provider.toUpperCase()
+  }
+
+  getProviderText (provider: Provider) {
+    return `${this.getProviderName(provider)} ${getLocale('for')}`
+  }
+
+  render () {
+    const { id, provider, onToggleTips, tipsEnabled } = this.props
+
```

Comment:
I would move provider check here and return null. So that we don't create empty space with `StyledEnableTipsSection` which has quite big padding

Reply:
++